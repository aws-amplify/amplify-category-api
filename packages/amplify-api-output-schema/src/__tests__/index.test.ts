import { versionedApiOutputSchema } from '..';

describe('Output Schema', () => {
  test('parses valid schema', () => {
    expect(() =>
      versionedApiOutputSchema.parse({
        version: '1',
        payload: {
          awsAppsyncRegion: 'us-east-1',
          awsAppsyncApiEndpoint: 'http://api',
          awsAppsyncAuthenticationType: 'api-key',
          awsAppsyncApiKey: 'fake-api-key',
        },
      }),
    ).not.toThrow();
  });

  test('throws error when parsing invalid schema', () => {
    expect(() => versionedApiOutputSchema.parse(12)).toThrowErrorMatchingSnapshot();
  });

  test('throws error when missing required paramse', () => {
    expect(() =>
      versionedApiOutputSchema.parse({
        version: '1',
        payload: {
          awsAppsyncRegion: 'us-east-1',
          // awsAppsyncApiEndpoint: 'http://api',
          awsAppsyncAuthenticationType: 'api-key',
          awsAppsyncApiKey: 'fake-api-key',
        },
      }),
    ).toThrowErrorMatchingSnapshot();
  });

  test('throws version is not matching', () => {
    expect(() =>
      versionedApiOutputSchema.parse({
        version: '2',
        payload: {
          awsAppsyncRegion: 'us-east-1',
          awsAppsyncApiEndpoint: 'http://api',
          awsAppsyncAuthenticationType: 'api-key',
          awsAppsyncApiKey: 'fake-api-key',
        },
      }),
    ).toThrowErrorMatchingSnapshot();
  });
});
