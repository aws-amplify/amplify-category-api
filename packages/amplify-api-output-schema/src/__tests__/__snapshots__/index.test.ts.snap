// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output Schema throws error when missing required paramse 1`] = `
"[
  {
    \\"code\\": \\"invalid_type\\",
    \\"expected\\": \\"string\\",
    \\"received\\": \\"undefined\\",
    \\"path\\": [
      \\"payload\\",
      \\"awsAppsyncApiEndpoint\\"
    ],
    \\"message\\": \\"Required\\"
  }
]"
`;

exports[`Output Schema throws error when parsing invalid schema 1`] = `
"[
  {
    \\"code\\": \\"invalid_type\\",
    \\"expected\\": \\"object\\",
    \\"received\\": \\"number\\",
    \\"path\\": [],
    \\"message\\": \\"Expected object, received number\\"
  }
]"
`;

exports[`Output Schema throws version is not matching 1`] = `
"[
  {
    \\"code\\": \\"invalid_union_discriminator\\",
    \\"options\\": [
      \\"1\\"
    ],
    \\"path\\": [
      \\"version\\"
    ],
    \\"message\\": \\"Invalid discriminator value. Expected '1'\\"
  }
]"
`;
